### PR TITLE
Fixes #2627 no expression profiles created

### LINKED ISSUE #2627: No expression profiles created when exporting a simulation from PK-Sim to MoBi.
When importing a pkml created with V11.2, no expression profiles are created. Though it is an expected behavior, expression values should still be somewhere, e.g. in the PSV.

### PR DESCRIPTION
**Pull request by msevestre**
Fixes #2627 

# Description

Ensures that we add the missing building blocks to the simulation

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Integration tests
- [ ] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/PKSim.Core/Model/Simulation.cs [hunk 1]
```diff
@@ -531,10 +531,7 @@ public override void AcceptVisitor(IVisitor visitor)
       /// <typeparam name="TBuildingBlock">type of the building blocks we are looking for in the simulation</typeparam>
       public virtual IEnumerable<TBuildingBlock> AllBuildingBlocks<TBuildingBlock>() where TBuildingBlock : class, IPKSimBuildingBlock
       {
-         return from usedBb in UsedBuildingBlocks
-            let bb = usedBb.BuildingBlock as TBuildingBlock
-            where bb != null
-            select bb;
+         return UsedBuildingBlocks.Select(x => x.BuildingBlock as TBuildingBlock).Where(x => x != null);
```
Comment (msevestre) [inline src/PKSim.Core/Model/Simulation.cs:534]: sometimes less is just more

File: tests/PKSim.Tests/ProjectConverter/v12/Converter11To12Specs.cs [hunk 2]
```diff
@@ -52,4 +52,23 @@ private void validateIndividual(Individual individual)
...
+         base.GlobalContext();
+         LoadProject("v11_expression_profile");
+         _allSimulations = All<Simulation>();
+         _allSimulations.Each(Load);
+      }
+
+      [Observation]
+      public void should_have_added_the_expression_used_by_the_individual_to_the_simulation()
+      {
+         _allSimulations.Each(s => s.AllBuildingBlocks<ExpressionProfile>().Count().ShouldBeGreaterThan(0));
```
Comment (msevestre) [inline tests/PKSim.Tests/ProjectConverter/v12/Converter11To12Specs.cs:71]: would fail before fix

**Review by msevestre — COMMENTED** [review]
The only issue with this is the assumption that the building block where not removed from the individual. I think it's pretty safe to assume it's ok

File: src/PKSim.Infrastructure/ProjectConverter/v12/Converter11To12.cs [hunk 3]
```diff
@@ -152,6 +161,26 @@ private void convertIndividual(Individual individual)
       private void convertSimulation(Simulation simulation)
       {
          convertIndividual(simulation.BuildingBlock<Individual>());
+
+         addMissingExpressionBuildingBlock(simulation);
+      }
+
+      private void addMissingExpressionBuildingBlock(Simulation simulation)
+      {
+         //in v11, the usage of the expression building block was not saved in the used building block. We need to add it if it's not there already
```
Comment (rwmcintosh) [inline src/PKSim.Infrastructure/ProjectConverter/v12/Converter11To12.cs:170]: This comment seems disconnected from the code. You might even want to place this comment on line 164.

File: src/PKSim.Core/Model/Simulation.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/PKSim.Core/Model/Simulation.cs:534]: and too much is just enough

**Review by rwmcintosh — APPROVED** [review]

**Pull request merged**